### PR TITLE
[clang][analyzer] PointerSubChecker should not warn on pointers converted to numerical type

### PR DESCRIPTION
**Pull request by balazske**
Pointer values casted to integer (non-pointer) type should be able to be subtracted as usual.

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/PointerSubChecker.cpp [hunk 1]
```diff
@@ -61,6 +61,10 @@ void PointerSubChecker::checkPreStmt(const BinaryOperator *B,
   if (LR->getSymbolicBase() || RR->getSymbolicBase())
     return;
 
+  if (!B->getLHS()->getType()->isPointerType() ||
+      !B->getRHS()->getType()->isPointerType())
+    return;
```
Comment (steakhal) [inline clang/lib/StaticAnalyzer/Checkers/PointerSubChecker.cpp:66]: Are sugared types passing this? Such as a typedef hiding the pointer type?
```c++
using ptr = int*;
```

Shouldnt we prefer canonical types instead?

Comment (balazske) [reply clang/lib/StaticAnalyzer/Checkers/PointerSubChecker.cpp:66]: The function `isPointerType` uses already the canonical type.

Comment (steakhal) [reply clang/lib/StaticAnalyzer/Checkers/PointerSubChecker.cpp:66]: Awesome!

**Review by NagyDonat — APPROVED** [review]
The change LGTM, I think we can merge it.

My only nitpick was that I tweaked the PR title to make it a bit more accurate. (Feel free to tweak it further / discuss this if you disagree with my choice.)

**Comment by karka228** [comment]
> The change LGTM, I think we can merge it.

I guess the testcase pointer-sub.c have to be fixed to work on the windows buildbot, also. Right?

**Comment by NagyDonat** [comment]
> > The change LGTM, I think we can merge it.
> 
> I guess the testcase pointer-sub.c have to be fixed to work on the windows buildbot, also. Right?

Right, thanks for catching my mistake!

**Pull request merged**